tailcfg/tailcfg.go

```diff
@@ -2653,6 +2653,17 @@ const (
 	// NodeAttrTrafficSteering configures the node to use the traffic
 	// steering subsystem for via routes. See tailscale/corp#29966.
 	NodeAttrTrafficSteering NodeCapability = "traffic-steering"
+
+	// NodeAttrTailnetDisplayName is how the tailnet name should be displayed to the user.
+	// Currently if this is set by a user this will only include letters, numbers,
+	// apostrophe, spaces, and hyphens. The only time this is not true,
+	// is if it defaults to the SID.
+	// i.e. "Tailscale - Dev", "User's Tailnet", or if it is a default SID
+	// "user@example.com"
+	// If a user manually set a display name the value that will be set here.
+	// If a user has not manually set a display name or has removed a display name
```

Comment:
I think we need to support this being empty/absent for old/alternate control plane servers. You can't assume this is populated in the client.